drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -362,30 +362,31 @@ static inline void wait_for_write_complete(const mtd_spi_nor_t *dev, uint32_t us
         i++;
 #if IS_USED(MODULE_ZTIMER_USEC)
         if (us) {
-            ztimer_sleep(ZTIMER_USEC, us);
+            uint32_t wait_us = us / div;
+            uint32_t wait_min = 2;
+
+            wait_us = wait_us > wait_min ? wait_us : wait_min;
+
+            ztimer_sleep(ZTIMER_USEC, wait_us);
             /* reduce the waiting time quickly if the estimate was too short,
              * but still avoid busy (yield) waiting */
-            if (us > 2) {
-                us -= (us / div);
-                div++;
-            }
+            div++;
         }
         else {
             j++;
             thread_yield();
         }
 #elif IS_USED(MODULE_XTIMER)
         if (us) {
-            xtimer_usleep(us);
+            uint32_t wait_us = us / div;
+            uint32_t wait_min = 2 * XTIMER_BACKOFF;
```

Comment:
I think this value should be at least 100 ` max(100,2 * XTIMER_BACKOFF)`

Reply:
Lets leave that for another one